spec/requests/middleware_deployments_spec.rb

```diff
@@ -0,0 +1,65 @@
+describe 'Middleware Deployments API' do
+  let(:deployment) { FactoryGirl.create(:middleware_deployment) }
+
+  def match_compressed(field)
+    eq ApplicationRecord.compress_id(deployment.send(field)).to_s
+  end
+
+  describe '/' do
+    it 'forbids access without an appropriate role' do
+      api_basic_authorize
+
+      run_get api_middleware_deployments_url
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'returns an empty listing of deployments' do
+      api_basic_authorize collection_action_identifier(:middleware_deployments, :read, :get)
+
+      run_get api_middleware_deployments_url
+
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to eq(
+        'name'      => 'middleware_deployments',
+        'count'     => 0,
+        'resources' => [],
+        'subcount'  => 0
+      )
+    end
+
+    it 'returns a a listing of deployments' do
+      deployment
+
+      api_basic_authorize collection_action_identifier(:middleware_deployments, :read, :get)
+
+      run_get api_middleware_deployments_url
+
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to eq(
+        'name'      => 'middleware_deployments',
+        'count'     => 1,
+        'resources' => [{
+          'href' => api_middleware_deployment_url(nil, deployment.compressed_id)
+        }],
+        'subcount'  => 1
+      )
+    end
+  end
+
+  describe '/:id' do
+    it 'returns the attributes of one deployment' do
+      api_basic_authorize
```

Comment:
individual resource gets, /api/_collection_/id will need the proper role identified

Reply:
and the equivalent forbidden tests for each